app.py

```diff
@@ -200,32 +205,35 @@ def make_preprocessor(queue=None):
                 url = await IOLoop.current().run_in_executor(
                     None, storage.copy, storage.QUARANTINE, storage.PERM, payload_id
                 )
-                produce_queue.append(
-                    {
-                        'topic': 'platform.upload.available',
-                        'msg': {
-                            'id': data.get('id'),
-                            'url': url,
-                            'service': data.get('service'),
-                            'payload_id': payload_id,
-                            'account': data.get('account'),
-                            'principal': data.get('principal'),
-                            'b64_identity': data.get('b64_identity'),
-                            'rh_account': data.get('account'),  # deprecated key, temp for backward compatibility
-                            'rh_principal': data.get('principal'),  # deprecated key, temp for backward compatibility
-                        }
+                data = {
+                    'topic': 'platform.upload.available',
+                    'msg': {
+                        'id': data.get('id'),
+                        'url': url,
+                        'service': data.get('service'),
+                        'payload_id': payload_id,
+                        'account': data.get('account'),
+                        'principal': data.get('principal'),
+                        'b64_identity': data.get('b64_identity'),
+                        'rh_account': data.get('account'),  # deprecated key, temp for backward compatibility
+                        'rh_principal': data.get('principal'),  # deprecated key, temp for backward compatibility
                     }
+                }
+                produce_queue.append(data)
+                logger.info(
+                    "data for topic [%s], payload_id [%s] put on produce queue (qsize now: %d): %s",
+                    data['topic'], payload_id, len(produce_queue), data
```

Comment:
Do you think printing the entire data dict in the logs could get kind of noisy?  Could we just print what we think might be relevant out of it?

Reply:
Sure, what if we only log full data dict as a debug log msg? By default log level is INFO so it shouldn't clog things then. See latest commit.

Reply:
that sounds good to me :)